euclidean_gcd/EuclideanGCD.java

```diff
@@ -0,0 +1,33 @@
+/*
+* first --> First number
+* second --> Second number
+* There are two implementations:
+* Recursive(euclideanGCDRecursive) and Non-Recursive(euclideanGCD)
+*/
+
+public class EuclideanGCD {
+    static int euclideanGCD(int first, int second) {
+        while(second != 0) {        // Iterate till second becomes zero
+            int temp = second;      // Temporary variable to hold value of second 
+            second = first % second;
+            first = temp;
+        }
+        return first;       // When second becomes 0, first becomes gcd of both
+    }
+
+    static int euclideanGCDRecursive(int first, int second) {
+        if(second == 0)
+            return first;       // First becomes gcd if second becomes zero
+        else
+            return euclideanGCDRecursive(second, (first % second));
```

Comment:
Readability depends on the person. For me, the ternary operators way is more acceptable and I will be able to read and grasp it faster, when reading someone else's code. YMMV. 
However, obviously this advantage only applies when ternary is not complex, like just one level. In more nested situations, if..else is the obvious choice.